### ISSUE TITLE
Not able to access the mounted drive

### ISSUE THREAD

**Issue by mspatil32** [GSConnect/gnome-shell-extension-gsconnect#1610]
### Describe the bug

Hi,

I think the app in phone got updated recently, and I'm getting the below error when trying to access the mounted drive in linux.

Please prove solutions.

![image](https://user-images.githubusercontent.com/105443715/234292939-5e0a9c99-908e-40fb-b681-f85d59938141.png)

### GSConnect version

55

### Installed from

GNOME Extensions website

**Comment by f-fouad** [comment]
You don't have permissions to access `/` of your phone.

Press `Ctrl+L` and append to the address `/storage/emulated/0` to access Android Internal storage. It should be the default.

**Comment by mspatil32** [comment]
Hi @f-fouad ,

Thanks for your reply.

Please tell me where should I add the address "/storage/emulated/0".

This issue came up after the "kdeconnect" app got updated yesterday in android. As a work around, I installed the older version of the app (from Jan 2023), and it is working fine.

**Comment by seanms** [comment]
I'm seeing the same problem after upgrading kdeconnect on my phone.

@mspatil32 When you're on the Nautlius page giving you the error, pressing Ctrl+L opens the address bar at the top.  Append "/storage..." to the address.

**Comment by mspatil32** [comment]
@seanms 
Oh okay, understood now. Does this help? Is the entire phone's filesystem accessible by this method? Previously, the Downloads folders in the internal storage was not accessible.

**Comment by seanms** [comment]
Yes, this appears to work around the problem.  I'm able to read and write the phone's file system.

**Comment by mspatil32** [comment]
Hi @seanms 
Thanks a lot. Will try it.
Currently, I installed the older version of the app (Jan 2023), and it doesn't give this problem, and I can access some of the folders in the phone (not the main internal storage and the Downloads folder).

**Comment by alexanderadam** [comment]
I'm seeing the same with recent versions and appending `/storage/emulated/0` helps indeed.
@andyholmes would it be possible to optionally this path via settings automatically?

**Comment by andyholmes** [comment]
No, it's the responsibility of the remote device to report the correct path. This doesn't relate to any settings in GSConnect.

**Comment by andyholmes** [comment]
Correct, the remote device must communicate the path. I haven't seen such a bug report yet, nope :/

**Comment by mspatil32** [comment]
Hi
Will there be an update to kde connect app to solve this issue?

**Comment by andyholmes** [comment]
You will have to open an upstream bug at https://bugs.kde.org

**Comment by mspatil32** [comment]
Sure. 
Thanks @andyholmes .

**Comment by alexanderadam** [comment]
Not sure whether it's related but I asked over [here](https://bugs.kde.org/show_bug.cgi?id=419897#c23). :man_shrugging:

**Comment by alexanderadam** [comment]
@andyholmes would it be possible just to add a configurable default path in the settings until the issue is fixed? :pleading_face:

**Comment by andyholmes** [comment]
No, because it's not a setting. It would be like adding a setting for payload transfer ports, when those should clearly be set internally by the device.

This setting is exposed on the device with the SSH server, because that device has access and control over the server. The remote device can't choose what the correct path is, it can only be told what it is.

**Comment by ajpy5215** [comment]
Hi
Here are the results of some experiments I have done :

First Configuration :
- Smartphone : Samsung A52S, Android 12, KDE Connect V1.24.5,
- PC : Ubuntu 20.04.6, Gnome V3.36.9, GSConnect V53 loaded from https://extensions.gnome.org/
Result : unable to access phone file system (no permission to access "/") from PC.
Appending "/storage/emulated/0" to the address in Nautilus with the "Ctrl-L" keys combination gives access to the phone file system.

Second configuration :
- Smartphone :  same as above,
- PC : same hardware as above, Ubuntu 22.04.2, Gnome V42.2, GSConnect V50 installed from Ubuntu software repository or loaded from https://extensions.gnome.org/ (V53 not available for this version of Gnome).
Result : the Phone connects to the PC but it’s impossible to mount it for accessing its file system – no error reported. Other functions seem to work.

Third configuration :
- Smartphone : same as above,
- PC : live session  Kubuntu 22.04.2 on the same PC hardware, KDE Connect (default installation)
Result : access to phone file system without any problem.

Log extract (slightly modified):

GSConnect: 53 (user)
GJS:       16405
Session:   x11
OS:        Ubuntu 20.04.6 LTS

...

mai 29 18:37:46 org.gnome.Shell.Extensions.GSConnect[24701]: [/service/plugins/sftp.js:_listDirectories/iter</<:160]: SM-A528B: Access denied
                                                             _listDirectories/iter</<@/home/xxxx/.local/share/gnome-shell/extensions/gsconnect@andyholmes.github.io/service/plugins/sftp.js:160:38

**Comment by alexanderadam** [comment]
So if KDE Connect works it is likely another issue?

**Comment by andyholmes** [comment]
@ajpy5215 Thanks for the logs!

The function appears to be a part of a refactoring in #831 which addressed #478. So in fact GSConnect generally ignores the path sent by the Android device and tries to list the directories itself.

This could be fixed by either reverting that patch, or possibly falling back to the advertised paths if the listing fails. Patches for either would be okay by me.

**Comment by alexanderadam** [comment]
So `/storage/emulated/0` is the main storage. How is the path to the external storage then? It seems that `/storage/emulated/1` doesn't work.

**Comment by ajpy5215** [comment]
I my case I get access to the external Sdcard by appending "storage/635F-151F" to "sftp://192.168.1.24:1739/".
Unfortunately the value "635F-151F" is specific to my Android/sdcard combination and will be different in other setups so cannot use that value on your side. I obtained the required value while experimenting with Kubuntu/KDEConnect. I’m not aware of other methods to get that value.
Note : appending " sdcard" gives the same result as appending "storage/emulated/0".

**Comment by neural75** [comment]
Same issue here with ubuntu 23.04. 
Is there a way to work around this issue? I've tried to bookmark the sftp url on the working /sdcard sub-path but obviously the address and port combination keeps changing, so it won't work for very long.
Any other way to create an alias( or something) automatically or from a scripts? (I don't know gvfs very much, whether it has some cmd line tool to list mount points or not, besides the nautilus interface).
My problem is that I will likely forget to add /sdcard as the good path (and the emulated thing even more).

**Comment by neural75** [comment]
> Same issue here with ubuntu 23.04. Is there a way to work around this issue? I've tried to bookmark the sftp url on the working /sdcard sub-path but obviously the address and port combination keeps changing, so it won't work for very long. 

I quote myself, I think I've found a simple work-around to share:
the mount point is available here: /run/user/1000/gsconnect/by-name/<device name>/sdcard 
device name is "Mi 9T Pro" in my  case, so I bookmarked that path under Nautilus as "Mi 9T Pro sdcard".
The path /run/user/1000/gsconnect/by-name/Mi 9T Pro/sdcard seems to be constant and working even when the sftp:// url switches port number.

**Comment by fjueic** [comment]
hi @andyholmes ,
I have been messing with source code for a few days.
I have an idea for the workaround.

When mounting storage I tried to create a bookmark with the device name which contains soft links representing internal storage and SD card. It seems like resolving network locations takes time leading me to experience a significant speed drop. I dropped this idea.

i saw some Stackoverflow code(can't find the link rn). What it did was, give the location of the nautilus uri of all instances every time it changed. i plan to use that program to monitor D-bus signal and monitor if I open some dummy folders representing the phone's internal storage and sd card.
I have the high-level image of how i wil do it.

at the end, it is just a workaround.
if such pr i can be accepted, i will try to integrate it into the code.
Otherwise, i will write all logic in Python and run a service as I'm comfortable in it as compared to JavaScript.

**Comment by andyholmes** [comment]
I'm not sure I understand your proposed solution? Nautilus really doesn't know anything about the remote storage, that either has to be sent by kdeconnect-android or queried with GVfs.

**Comment by andyholmes** [comment]
Ah, yeah that seems out-of-scope for GSConnect. It might make sense as a Nautilus plugin though.

Probably kdeconnect-android should just serve the path to the SD Card if this is an issue though.

**Comment by fjueic** [comment]
i was checking data shared by kde-connect and it turns out it servers paths for internal storage and sd card.
issue lies in `with gvfs we can only mount the location provided by whatever is servering sftp.`

**Comment by fjueic** [comment]
kde plasma implements sompthing like `kdeconnect:///device-name`  with internal storage and sd card. And these just redirects to to mapped uri like `/storage/emulated/0` and `/stoarge/emulated/sdcardvalue`.
i dont really know anything about custom uri. but i got a workaround(lol)

**Comment by fjueic** [comment]
@andyholmes please confirm that such PR will not be accepted

**Comment by andyholmes** [comment]
I would not accept a PR that tries to address issues in GVfs or Nautilus.

> The function appears to be a part of a refactoring in #831 which addressed #478. So in fact GSConnect generally ignores the path sent by the Android device and tries to list the directories itself.
> 
> This could be fixed by either reverting that patch, or possibly falling back to the advertised paths if the listing fails. Patches for either would be okay by me.

I would accept a PR that did either of these two things.

**Comment by fjueic** [comment]
If anyone in future is interested in a workaround check [this repo](https://github.com/fjueic/gsconnect-mount-manager). It also handles SD card.
i tested in multiple live environment like manjaro, garuda Linux, Pop OS.
![example](https://github.com/GSConnect/gnome-shell-extension-gsconnect/assets/89282645/3f833bcb-4220-46ef-ad05-c2aec999306e)

**Comment by AyushSehrawat** [comment]
Facing same issue on PopOS with gnome 42.9, fixed by appending the path

**Comment by RushingAlien** [comment]
https://github.com/GSConnect/gnome-shell-extension-gsconnect/issues/1203#issuecomment-1886359637
> I've read the KDE Connection Android app code, and I noticed that KDE Connection sends multiple paths, but GSConnect only mounts the root directory; Here:

**Comment by andyholmes** [comment]
> I would not accept a PR that tries to address issues in GVfs or Nautilus.
> 
> > The function appears to be a part of a refactoring in #831 which addressed #478. So in fact GSConnect generally ignores the path sent by the Android device and tries to list the directories itself.
> > This could be fixed by either reverting that patch, or possibly falling back to the advertised paths if the listing fails. Patches for either would be okay by me.
> 
> I would accept a PR that did either of these two things.

**Comment by nebulosa2007** [comment]
After investigating exactly what [gsconnect-mount-manager](https://github.com/fjueic/gsconnect-mount-manager) does, I found the easiest way to fix the mount path for Nautilus without installing anything.

Steps:
1. Mount your phone as usual, open the mounted folder in Nautilus and get the error as in the first comment.
2. Press `Ctrl+L` and copy the path, e.g. `sftp://192.168.1.238:1761/`
3. Edit the Nautilus bookmarks file `gedit ~/.config/gtk-3.0/bookmarks`
4. Add a line like `sftp://192.168.1.238:1761/storage/emulated/0 MyPhone`, save and exit.
5. `MyPhone` bookmark in Nautilus should appear, which should open the phone folder without error on click.

This bookmark will only work if the phone has been mounted with GSConnect. To work independently without GSC mount, you need to make a `~/.ssh/config` file, where you should specify the ssh connection parameters (In progress...).

**Comment by fjueic** [comment]
> After investigating exactly what [gsconnect-mount-manager](https://github.com/fjueic/gsconnect-mount-manager) does

u could have tagged me. [gsconnect-mount-manager](https://github.com/fjueic/gsconnect-mount-manager) is a lil thing I wrote when I knew nothing about linux interfaces.

> Add a line like sftp://192.168.1.238:1761/storage/emulated/0 MyPhone, save and exit.

what if IP address changes? for me when i change classroom in university.
this case is handled in [gsconnect-mount-manager](https://github.com/fjueic/gsconnect-mount-manager)

if there are and question regarding the working of `gsconnect-mount-manager` just ask me or if you want me to explain everything i can do that too

**Comment by juanmah** [comment]
> 4. Add a line like `sftp://192.168.1.238:1761/storage/emulated/0 MyPhone`, save and exit.

The port is dynamic, and it's changing at each connection.

> This bookmark will only work if the phone has been mounted with GSConnect. To work independently without GSC mount, you need to make a `~/.ssh/config` file, where you should specify the ssh connection parameters (In progress...).

I'm interested in knowing how to connect without GSC.

**Comment by fjueic** [comment]
> I'm interested in knowing how to connect without GSC.

how I did it before [gsconnect-mount-manager](https://github.com/fjueic/gsconnect-mount-manager) 

using sshfs with the termux app on Android. I had some python code running in termux and laptop to handle dynamic ip

**Comment by fjueic** [comment]
> Your mount implementation

Mounting is handled by GSConnect.

What I tried to do was handle the paths.

I will briefly explain what my implementation is doing:

- `/` is not accessible in Android (unless your device is rooted, I guess) and we need to access `/storage/emulated/0` to access the storage.
> It turns out it serves paths for internal storage and SD card.
> The issue lies with GVFS; we can only mount the location provided by whatever is serving SFTP.

### Executing the install script:
- The install script edits your GSConnect file to add two extra lines of code in [sftp.js](https://github.com/GSConnect/gnome-shell-extension-gsconnect/blob/9d274631033cf627580e274f953e8b03b55a9c08/src/service/plugins/sftp.js), both wrapped in try and catch. THAT IS IT, nothing else is edited in your GSConnect installation.
- If something were to go wrong in my implementation, I don't see it causing any issues with the working of GSConnect.
- Plus, a service is added (functionality explained later).
- [A file](https://github.com/fjueic/gsconnect-mount-manager/blob/main/mountInfoHandler.js) containing my function is put along with GSConnect.

### On Mounting:
- The path data (device, SD card) that we get from the Android device is then put in dconf of GSConnect.
- A directory representing the device is created in `~/.gsconnectMounts/`. In the [example](https://github.com/fjueic/gsconnect-mount-manager/blob/main/example.gif), it's `realme 8`.
   - Links pointing to the parent directory are created, with each link representing different storage media connected to your Android device. In the [example](https://github.com/fjueic/gsconnect-mount-manager/blob/main/example.gif), it's `Internal shared Storage` and `SD card`.
   #### - All this is just dummy directories
![image](https://github.com/GSConnect/gnome-shell-extension-gsconnect/assets/89282645/d1910774-8b1d-4653-9b9b-e3f2c4da2540)
- And finally, it's just bookmarked.

### Accessing:
- A Python service listens to D-Bus events of Nautilus.
- Whenever a user accesses directories in `~/.gsconnectMounts`, info about that device IP and port and path is read from dconf, the correct path is built, and a new Nautilus window is opened.

**Comment by fjueic** [comment]
> - The install script edits your GSConnect file to add two extra lines of code in [sftp.js](https://github.com/GSConnect/gnome-shell-extension-gsconnect/blob/9d274631033cf627580e274f953e8b03b55a9c08/src/service/plugins/sftp.js), both wrapped in try and catch. THAT IS IT, nothing else is edited in your GSConnect installation.
> - If something were to go wrong in my implementation, I don't see it causing any issues with the working of GSConnect.

My main focus was not to interfere with gsconnect working.

> Spent two evenings debugging

- disable the service
```
systemctl --user stop gsconnect-mount-manager.service
systemctl --user disable gsconnect-mount-manager.service
```

reinstall the gsconnect to remove those edit two extra lines of code in [sftp.js](https://github.com/GSConnect/gnome-shell-extension-gsconnect/blob/9d274631033cf627580e274f953e8b03b55a9c08/src/service/plugins/sftp.js), both wrapped in try and catch.

if this does not fix your issue it's most probably not due to my implementation. I may be completely wrong, so if you can point me to a potential issue I would appreciate it.

**Comment by nebulosa2007** [comment]
> I may be completely wrong, so if you can point me to a potential issue I would appreciate it.

Found simple solution: `rm $HOME/.config/dconf/user` Now mounting work as expected but again with wrong path in Nautilus.

Reverting to making proper bookmark, we could use two ways: 
* Read actions from `dconf` to get dynamic IP and port
* Just use `ps aux` after GSConnect mounting and adding bookmark:
```console
mf="storage/emulated/0 MyPhone"; 
ps aux|awk -v r="$mf" '/ sftp/{printf("%s://%s:%s/%s\n",$31,$30,$28,r)}'| head -1 >> .config/gtk-3.0/bookmarks
```
Add some lines to handle the bookmark update, make a keybinding, et voila!

Next step: using ~/.ssh/config. 
In my home network, the phone has a dns name `<phonename>.lan`, for example `pixel-4.lan`. Another situation, the phone does not have a dns name, but the router often gives the IP for the phone the same if that phone appears frequently on the wi-fi network. Get the IP or dns of the phone anyway, then:

1. Add line `sftp://pixel-4.lan/storage/emulated/0 Pixel` in `.config/gtk-3.0/bookmarks`
2. Add options for ssh connection in `~/.ssh/config`:
```console
Host pixel-4.lan
  Port 1739
  User kdeconnect
  IdentityFile ~/.config/gsconnect/private.pem
  IdentitiesOnly yes
  IdentityAgent none
  PubkeyAcceptedKeyTypes ssh-rsa
  HostKeyAlgorithms +ssh-rsa
  ForwardX11 no
  ForwardAgent no
  PermitLocalCommand no
  ClearAllForwardings yes
  Protocol 2
  NoHostAuthenticationForLocalhost yes
  ControlMaster auto
  ControlPath=/run/user/1000/gvfsd-sftp/%C
```

Nautilus will open the phone folder on click. GSConnect should be loaded in the background, of course.

**Comment by fjueic** [comment]
> Won't help much but :
> 
> ⋅ sftp://IP:portA/storage/emulated/**0** → seems to be the first user created on Android ; ⋅ sftp://IP:portB/storage/emulated/**10** → …the second user.

i don't think anybody is using multiple user is android

[dynamically handled here](https://github.com/fjueic/gsconnect-mount-manager)

**Comment by Coeur-Noir** [comment]
Well I do that's how I know...

:-D
-- 
Envoyé depuis Motorola g84 via le courrieleur libre K-9 Mail ( prochainement Thunderbird ).

Le 3 mai 2024 11:06:12 GMT+02:00, fjueic ***@***.***> a écrit :

**Comment by akvedi** [comment]
There is a workaround for this. When gsconnect mounts the sftp it adds a new folder in the `/run/user/1000/gvfs` with `host ip` and `port` as a folder name. <br>

<img src="https://github.com/GSConnect/gnome-shell-extension-gsconnect/assets/75240610/b60c3e19-290c-4318-848d-f3b6747372fd" alt="ss" width="150"/> <br>

I wrote the below bash script to automatically find the `host ip` and `port` from the folder name and then open Nautilus with the correct path.


```bash
#!/bin/bash
# A script to open GSconnect sftp folder with correct path

# Define the path to the GVFS folder
gvfs_path="/run/user/1000/gvfs"

# Find the folder that matches the pattern "sftp:host=*,port=*"
folder=$(find "$gvfs_path" -type d -name "sftp:host=*,port=*" | head -n 1)

if [ -z "$folder" ]; then
  echo "No matching folder found."
  exit 1
fi

# Extract host and port from the folder name
regex="sftp:host=([^,]*),port=([0-9]*)"
if [[ $folder =~ $regex ]]; then
  host=${BASH_REMATCH[1]}
  port=${BASH_REMATCH[2]}
else
  echo "Failed to extract host and port from folder name."
  exit 1
fi

# Construct the sftp URL
sftp_url="sftp://$host:$port/storage/emulated/0"

echo "Opening $sftp_url in Nautilus..."

# Open the sftp URL in Nautilus
nautilus "$sftp_url" &

```

<br>

This script works with Nautilus but can be easily modified to work with other file managers.

**Comment by Froggy232** [comment]
Hi,
Is there any news on this? Is it really not possible to modify the gsconnect behaviour to automatically point to internal storage, as it seems kdeconnect won't change their position?
This way, it would be compatible with kdeconnect android without any modification or hacks.
Thanks for your work anyway, it's an incredible useful extension, even with this little bug. Also, maybe it would be a good idea to at least disable automatic sftp mounting, and maybe add a warning?
Thanks again

**Comment by piotrminkina** [comment]
Using `/sdcard` instead of `/storage/emulated/0` is much faster ;)

**Comment by ferdnyc** [comment]
@Froggy232 @piotrminkina 

It doesn't matter what path we use, due to the way gvfs works there is no way to mount any directory other than the root of a remote SFTP server. You have to navigate to the path on that server, which means hitting permission-denied errors. (See https://github.com/GSConnect/gnome-shell-extension-gsconnect/pull/1793#issuecomment-2620206990 and the discussions that follows for more.)

**Comment by juanmah** [comment]
> Using `/sdcard` instead of `/storage/emulated/0` is much faster ;)

I checked the throughput of the two paths, and, in my case, are practically the same.

**Comment by piotrminkina** [comment]
@ferdnyc You're right. My comment was not intended to point out a solution to the problem, but to make the problem easier to live with.

@juanmah I think _/sdcard_ is a symbolic link to _/storage/emulated/0_. By writing “much faster”, I meant that it would be much faster to type the former path instead of the latter.

**Comment by Coeur-Noir** [comment]
Hi,

...beware there is a path per user session on Android.

/0 is the first user, /10 a second one so the path might be

/storage/emulated/10


-- 
Envoyé depuis Motorola g84 via Thunderbird, client courriel libre et open-source.

Le 6 février 2025 11:48:11 GMT+01:00, Piotr Minkina ***@***.***> a écrit :

**Comment by SalvatoreNoschese** [comment]
Hi, no way to fix this issue?
looks like a very old bug and i really hope can be fixed asap.
i tried also to make a nautilus extension (py) to redirect 
sftp://192.168.1.***:****/
to:
sftp://192.168.1.***:****/storage/emulated/0
but atm without success.

ty.

**Comment by ferdnyc** [comment]
@SalvatoreNoschese 

> Hi, no way to fix this issue?

There is no way to fix this issue **ideally**, because of how GVFS works.

I have some ideas on how to fix it "as much as possible", as I titled my WIP PR for those changes. (#1921) Those ideas are in a comment at the bottom of the PR, they are not yet implemented in the PR.

I did make significant progress on a partially complete implementation, locally... which I then lost access to when my main coding system died two weeks ago. Since then I've been working crippled on my 12-year-old, 2-core dinosaur of a laptop, so it's been slow going.

I have the drive from the machine and at some point I'll get it mounted and recover access to all of my work in progress, but right now we're all focused on the next GSConnect release. It's hit some snags we're scrambling to fix, since it's already delayed and needs to be priority #ONLY for the moment.

> looks like a very old bug and i really hope can be fixed asap.

All of which is to say, "asap" is definitely off the table.

> i tried also to make a nautilus extension 

Yeah... as you discovered, while Nautilus is extensible, it's not "that kind" of extensible. 

What you'd really want, in an ideal world, is a GVFS plugin that implements a `gsconnect:` or `kdeconnect:` scheme -- that'd give you full control. But out-of-tree GVFS modules aren't supported, so the code would have to be accepted and maintained by the GNOME project.

**Comment by vjaykrsna** [comment]
### Tired of manually mounting your phone with GSConnect?

I created a simple, lightweight script that completely automates the process. It runs in the background and automatically:

-   **Mounts** your phone's storage when it connects.
-   **Creates a bookmark and a symlink** in your home directory using your device's name.
-   **Cleans everything up** when you disconnect.

It's a "set it and forget it" solution that makes accessing your phone's files on GNOME seamless.

---

**Check it out on GitHub:**
### [https://github.com/vjaykrsna/gsconnect-mount-manager](https://github.com/vjaykrsna/gsconnect-mount-manager)

If you find it useful, please consider giving the repo a ⭐ star!

If you run into any problems or have a feature request, don't hesitate to [open an issue](https://github.com/vjaykrsna/gsconnect-mount-manager/issues).

**Comment by fjueic** [comment]
> ### Tired of manually mounting your phone with GSConnect?
> 
> I created a simple, lightweight script that completely automates the process. It runs in the background and automatically:
> 
> -   **Mounts** your phone's storage when it connects.
> -   **Creates a bookmark and a symlink** in your home directory using your device's name.
> -   **Cleans everything up** when you disconnect.
> 
> It's a "set it and forget it" solution that makes accessing your phone's files on GNOME seamless.
> 
> ---
> 
> **Check it out on GitHub:**
> ### [https://github.com/vjaykrsna/gsconnect-mount-manager](https://github.com/vjaykrsna/gsconnect-mount-manager)
> 
> If you find it useful, please consider giving the repo a ⭐ star!
> 
> If you run into any problems or have a feature request, don't hesitate to [open an issue](https://github.com/vjaykrsna/gsconnect-mount-manager/issues).
> 

I made something similar, coincidentally the same name. 

Symbolic link is what came to my mind but speed is low compared to the normal path. The difference was huge enough to not use this approach. 

Is this solved?
If yes, how did you solve it?
If no, I think you should add this in read me.

**Comment by vjaykrsna** [comment]
> 
> Is this solved? If yes, how did you solve it? If no, I think you should add this in read me.

@fjueic 
yeah i saw your and forked first but I didn't like how it open internal storage folder in a new popup window
I tried foxing it but that's the default behaviour i wasn't able to change
so i chose to rewrite the whole thing lol

Yeah it works have tested with kde connect maybe for device name logic i have to make changes for kde connect
unless someone needs that i am not gonna put efforts

**Comment by fjueic** [comment]
> 
> > 
> > Is this solved? If yes, how did you solve it? If no, I think you should add this in read me.
> 
> @fjueic 
> yeah i saw your and forked first but I didn't like how it open internal storage folder in a new popup window
> I tried foxing it but that's the default behaviour i wasn't able to change
> so i chose to rewrite the whole thing lol
> 
> Yeah it works have tested with kde connect maybe for device name logic i have to make changes for kde connect
> unless someone needs that i am not gonna put efforts 

The main thing I wanted to ask was, if speed drops because of symbolic links.

**Comment by vjaykrsna** [comment]
> > > Is this solved? If yes, how did you solve it? If no, I think you should add this in read me.
> > 
> > 
> > [@fjueic](https://github.com/fjueic)
> > yeah i saw your and forked first but I didn't like how it open internal storage folder in a new popup window
> > I tried foxing it but that's the default behaviour i wasn't able to change
> > so i chose to rewrite the whole thing lol
> > Yeah it works have tested with kde connect maybe for device name logic i have to make changes for kde connect
> > unless someone needs that i am not gonna put efforts
> 
> The main thing I wanted to ask was, if speed drops because of symbolic links.

no it didn't
but i don't know why opening root folder in file manager some times crashes it
maybe i will try opening nautilus in terminal and then debug it

**Comment by fjueic** [comment]
> no it didn't

nah, you didn't test properly. Also it doesn't handle the sd card. You can definitely make more improvements.
Bookmark is fine as it is directly opens path but the symbolic link is not a solution.
Add the sd card support and this will good solution.

<img width="377" height="88" alt="Image" src="https://github.com/user-attachments/assets/82d1d6d2-fbdc-44ed-b2ce-4106ad28dd57" />

<img width="332" height="87" alt="Image" src="https://github.com/user-attachments/assets/ae735db2-3f49-420e-b242-1a6dd984c777" />

**Comment by vjaykrsna** [comment]
> > no it didn't
> 
> nah, you didn't test properly. Also it doesn't handle the sd card. You can definitely make more improvements. Bookmark is fine as it is directly opens path but the symbolic link is not a solution. Add the sd card support and this will good solution.
> 

Yeah i don't have a device with sdcard so i didn't implement it maybe in future I'll do that
And regarding the speed issue I use the symblink i added direct

**Comment by vjaykrsna** [comment]
> > no it didn't
> 
> nah, you didn't test properly. Also it doesn't handle the sd card. You can definitely make more improvements. Bookmark is fine as it is directly opens path but the symbolic link is not a solution. Add the sd card support and this will good solution.


i have added sdcard support and improved things alot can you try it out and let me know if that works
